components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt

```diff
@@ -41,10 +40,22 @@ fun AwesomeBar(
     onScroll: () -> Unit = {},
     profiler: Profiler? = null
 ) {
-    val groups = remember(providers) {
-        listOf(
+    val mapTitlesToProviders = HashMap<String?, ArrayList<AwesomeBar.SuggestionProvider>>()
+    for (provider in providers) {
+        var groupedProviders: ArrayList<AwesomeBar.SuggestionProvider>? =
+            mapTitlesToProviders[provider.groupTitle()]
+        if (groupedProviders == null) {
+            groupedProviders = ArrayList()
+        }
+        groupedProviders.add(provider)
+        mapTitlesToProviders[provider.groupTitle()] = groupedProviders
+    }
+    val groups: ArrayList<AwesomeBar.SuggestionProviderGroup> = ArrayList()
+    for (item in mapTitlesToProviders) {
+        groups.add(
             AwesomeBar.SuggestionProviderGroup(
-                providers = providers
+                providers = item.value,
+                title = item.key
             )
         )
```

Comment:
If the providers and the group titles don't change we could keep the old `remember` functionality to avoid recompositions with something like:

```suggestion
val groups = remember(providers) {
    providers
        .groupBy { it.groupTitle() }
        .map {
            AwesomeBar.SuggestionProviderGroup(
                providers = it.value,
                title = it.key
            )
        }
}
```

Reply:
Thank you @Mugurell !